File: app/src/utility/CaveNode.js
import * as CaveTiles from './CaveTiles';

class CaveNode {
  constructor (nodeIndex) {
    this.nodeIndex = nodeIndex;

    this.adjacentNodes = [];
    this.nodeTile = CaveTiles.EMPTY_TILE;
    this.ROOM = 'ROOM';
    this.WALL = 'WALL';
  }

  getNodeIndex () {
    return this.nodeIndex;
  }

  addAdjacentNode (node) {
    this.adjacentNodes.push(node);
  }

  getAdjacentNodes () {
    return this.adjacentNodes;
  }

  isNodeOccupied () {
    return this.nodeTile != CaveTiles.EMPTY_TILE;
  }

  markOccupiedWith (tile) {
    this.nodeTile = tile;
  }

  getNodeTileType () {
    return this.nodeTile;
  }
}

export default CaveNode;

File: src/main/java/com/example/caverna/model/CaveNode.java
package com.example.caverna.model;

import java.util.ArrayList;

public class CaveNode {
    private final Integer nodeIndex;
    private final ArrayList<CaveNode> adjacentNodes;
    private CaveTileEnum caveTile;

    public CaveNode(int nodeIndex) {
        this.nodeIndex = nodeIndex;

        adjacentNodes = new ArrayList<>();
        caveTile = CaveTileEnum.EMPTY;
    }

    public Integer getCaveNodeIndex() {
        return nodeIndex;
    }

    public void addAdjacentNode(CaveNode caveNode) {
        adjacentNodes.add(caveNode);
    }

    public ArrayList<CaveNode> getAdjacentNodes() {
        return adjacentNodes;
    }

    public Boolean isNodeOccupied() {
        return caveTile != CaveTileEnum.EMPTY;
    }

    public void markOccupiedWith(CaveTileEnum caveTile) {
        this.caveTile = caveTile;
    }

    public CaveTileEnum getCaveTile() {
        return caveTile;
    }
}
